plugins/alerting/pkg/alerting/api_conditions.go

```diff
@@ -515,60 +477,59 @@ func (p *Plugin) Timeline(ctx context.Context, req *alertingv1.TimelineRequest)
 	cortexStep := durationpb.New(req.LookbackWindow.AsDuration() / 500)
 	var wg sync.WaitGroup
 	var addMu sync.Mutex
-	for idx := range conditions {
-		idx := idx // capture in closure
+	for _, cond := range conditions {
```

Comment:
The synchronization logic here will have significant lock contention. I suggest refactoring this so that you don't need to lock in the goroutine, instead setting resp.Items all at once, at the end.